score.py

```diff
@@ -36,27 +37,70 @@ def score(gold, gold_col, pred, pred_col):
     }
 
 
-def main():
-    """Main function."""
-    args = get_args()
+def extract_gs_file(folder):
+    """Extract goldstandard file from folder."""
+    files = glob(os.path.join(folder, "*"))
+
+    # Filter out the manifest file
+    files = [f for f in files if os.path.basename(f) != 'SYNAPSE_METADATA_MANIFEST.tsv']
 
-    pred = pd.read_csv(args.predictions_file)
-    gold = pd.read_csv(args.goldstandard_file)
+    if len(files) != 1:
+        raise ValueError(
+            "Expected exactly one goldstandard file in folder. "
+            f"Got {len(files)}. Exiting."
+        )
+    return files[0]
 
-    # Replace spaces in column headers in case they're found.
-    gold.columns = [colname.replace(" ", "_") for colname in gold.columns]
 
-    # Strip leading and trailing spaces as needed.
-    gold = gold.map(lambda x: x.strip() if isinstance(x, str) else x)
-    pred = pred.map(lambda x: x.strip() if isinstance(x, str) else x)
+def check_validation_status(filename, args):
+    """
+    Determine whether the validate.py script successfully validated the data.
+    """
+    
+    with open(filename, "r") as f:
+        status_result = json.load(f)
+    print("Checking the validation_status.")
+    
+    if status_result.get("validation_status") == "INVALID":
+        new_data = {"validation_status": "INVALID",
+                    "validation_errors": status_result.get("validation_errors"),
+                    "score_status":  "INVALID",
+                    "score_errors": "Submission could not be evaluated due to validation errors."}
+    
+        with open(filename, "w") as out:
+            out.write(json.dumps(new_data))
+
+        print("INVALID")
+    else:
+        print("Scoring the submission.")
```

Comment:
This print statement might show up in the status of the submission, but we can test it to see if that's the case. For the PEGS challenge there were no other print statements in these scripts apart from the status update ([link](https://github.com/Sage-Bionetworks-Challenges/pegs-evaluation/blob/main/validate.py)). Everything else looks good to me!

Reply:
@jaymedina to clarify, when an error is encountered, the nf workflow knows to ignore the additional print statements but that's not the case when the submission is valid?

Just wanted to make sure I understand since in the sample scoring template, we do include [additional print statements](https://github.com/Sage-Bionetworks-Challenges/orca-evaluation-templates/blob/main/score.py#L119-L136) as well.

Reply:
I suspect that anything printed out to the console after running `score.py` or `validate.py` will be used as the `status` annotation on Synapse.

For example, [these](https://github.com/Sage-Bionetworks-Workflows/nf-synapse-challenge/blob/a96f0de05eda16ec54c4c645d9d62c88057f2341/modules/validate.nf#L15) are the expected outputs when executing the validation script, where `status` is a variable that captures the standard output of the command execution, so I'm suspecting this means it captures the entire output (all the print statements, rather than just the last one).

I think to stay on the safe side, we should remove all print statements from `score/validate.py` that are not the status updates themselves.